company-quickhelp.el

```diff
@@ -225,18 +225,13 @@ currently active `company' completion candidate."
                     (setq doc (pos-tip-truncate-string doc max-width max-height)
                           w-h (pos-tip-string-width-height doc))))
                   (let* ((str (propertize doc 'face 'company-quickhelp-face)))
-                    (pos-tip-show-no-propertize str fg-bg (overlay-start ovl) nil timeout
+                    (pos-tip-show-no-propertize str fg-bg pos nil timeout
                                                 (pos-tip-tooltip-width (car w-h) (window-font-width nil 'company-quickhelp-face))
                                                 ;;(pos-tip-tooltip-height (cdr w-h) (window-font-height nil 'company-quickhelp-face) frame) ;; height generally truncated short
                                                 nil ;; dummy height for testing
-                                                nil (+ overlay-width overlay-position) 1)))
+                                                nil (+ overlay-width overlay-position) dy)))
```

Comment:
The different dy offset gives the wrong height, at least in my build. By default the y pos is lower by 1, and the added 1 is used to fix that quirk. Am i missing something?

Reply:
Could you give an example? Maybe show a screenshot?

The 1 is not "added". The default value is the height of a line in pixels.

Reply:
I mean added in the sense that there is an offset of 1. Here's the example (with a face having a smaller height than the default face).

Using `dy` instead of 1:
![image](https://user-images.githubusercontent.com/37450282/85584289-4bda1600-b63f-11ea-8510-b7683e4d887f.png)


Using 1 as before:
![image](https://user-images.githubusercontent.com/37450282/85584033-12090f80-b63f-11ea-8f88-ffa9a806ebde.png)

Reply:
Also, using dy, there's a strange (or maybe that's what you wanted) behaviour with very long quickhelp tooltips. Instead of moving the by the excess the would have if staying in the default position, they get flipped like this.

![image](https://user-images.githubusercontent.com/37450282/85585878-b3449580-b640-11ea-9584-e9a16daf3d35.png)

While the expected behaviour should be this

![image](https://user-images.githubusercontent.com/37450282/85586117-eb4bd880-b640-11ea-8c25-768cbc39ba34.png)

In these images you can see the abnormal height applied by pos-tip that cuts the paragraphs short. For some reason using the real char height with 

````elisp
;; Tooltip height shorter than needed
(pos-tip-tooltip-height (cdr w-h) (window-font-height nil 'company-quickhelp-face) frame)
````
Creates this bug.

As opposed to using `frame-char-height` that just is the wrong character height in this case, and gives this result (as we said before).

![image](https://user-images.githubusercontent.com/37450282/85587004-ae341600-b641-11ea-9242-49aa8ce74b39.png)

The problem would be fixed if we could find a way to consistently get the frame char height and if/or pos-tip was able to consistently calculate the tooltip size in relation to the different char height. 

Another solution would be to just cut the text when it gets out of the screen (or just leave it be), but that would be evading the problem more than a solution.

Reply:
Thank you, I managed to reproduce the first example fine. Does the latest commit fix in master it for you?

Reply:
In the rest of the screenshots, are you describing the behavior from master, or from this branch?

Too-long (or wide) tooltips indeed can be unwieldy, but they generally get positioned the best pos-tip can do with the current code. Or is there a particular commit that made it broken for you?

Reply:
Oh, never mind. I reproduced it with the `semantic-default-submodes` variable's doc buffer. But whether I add 1 or not, it doesn't really matter.

Looks like a pos-tip bug.

Reply:
Is pos-tip still mantained? Or is it just obsolete?

Reply:
Either way, it should work better now. Please test. And thanks for the report and all the screenshots. :+1:

Reply:
> Is pos-tip still mantained? Or is it just obsolete?

Doesn't seem to be developed much. OTOH, there's just one open issue in the repo.